docs/spec/spec.md

```diff
@@ -0,0 +1,1031 @@
+that makes it easier to use, combine, and create network services.
+
+# Contents
+1. [Overview](#1-overview)
+2. [gRPC Command Line Interface (CLI)](#1-grpc-command-line-interface-cli)
+3. [Protocol Buffers to Ballerina Data Mapping](#3-protocol-buffers-to-ballerina-data-mapping)
+4. [gRPC Communication](#4-grpc-communication)
+   * 4.1. [Unary RPC](#41-unary-rpc)
+   * 4.2. [Server Streaming RPC](#42-server-streaming-rpc)
+   * 4.3. [Client Streaming RPC](#43-client-streaming-rpc)
+   * 4.4. [Bidirectional Streaming RPC](#44-bidirectional-streaming-rpc)
+5. [gRPC Security](#51-authentication-and-authorization)
+   * 5.1. [Authentication and Authorization](#51-authentication-and-authorization)
+      * 5.1.1. [Declarative Approach](#511-declarative-approach)
+         * 5.1.1.1. [Service - File User Store](#5111-service---file-user-store)
+         * 5.1.1.2. [Service - LDAP User Store](#5112-service---ldap-user-store)
+         * 5.1.1.3. [Service - JWT](#5113-service---jwt)
+         * 5.1.1.4. [Service - OAuth2](#5114-service---oauth2)
+         * 5.1.1.5. [Client - Basic Auth](#5115-client---basic-auth)
+         * 5.1.1.6. [Client - Bearer Token Auth](#5116-client---bearer-token-auth)
+         * 5.1.1.7. [Client - Self Signed JWT Auth](#5117-client---self-signed-jwt-auth)
+         * 5.1.1.8. [Client - OAuth2](#5118-client---oauth2)
+      * 5.1.2 [Imperative Approach](#512-imperative-approach)
+         * 5.1.2.1. [Service - File User Store](#5121-service---file-user-store)
+         * 5.1.2.2. [Service - LDAP User Store](#5122-service---ldap-user-store)
+         * 5.1.2.3. [Service - JWT](#5123-service---jwt)
+         * 5.1.2.4. [Service - OAuth2](#5124-service---oauth2)
+         * 5.1.2.5. [Client - Basic Auth](#5125-client---basic-auth)
+         * 5.1.2.6. [Client - Bearer Token Auth](#5126-client---bearer-token-auth)
+         * 5.1.2.7. [Client - Self Signed JWT Auth](#5127-client---self-signed-jwt-auth)
+         * 5.1.2.8. [Client - OAuth2](#5128-client---oauth2)
+6. [gRPC Utility Functions](#6-grpc-utility-functions)
+   * 6.1. [gRPC Deadline](#61-grpc-deadline)
+   * 6.2. [gRPC Compression](#62-grpc-compression)
+   * 6.3. [gRPC Access and Trace Logs](#63-grpc-access-and-trace-logs)
+   * 6.4. [gRPC Retry](#64-grpc-retry)
+
+
+# 1. Overview
+Ballerina gRPC standard library has five primary aspects in handling values.
+1. gRPC CLI (command line interface)
+2. Protocol buffers to Ballerina data mapping
+3. gRPC communication
+4. gRPC Security
+5. gRPC utility functions
+
+# 2. gRPC Command Line Interface (CLI)
+
+Ballerina language has a command-line interface that manages the lifecycle of a Ballerina program (such as build, test, and run). In addition, Ballerina CLI contains all the gRPC related stub and service skeleton generation capabilities. The gRPC command in Ballerina CLI is as follows.
+
+```sh
+bal grpc --input <proto-file-path> --output <output-directory> --mode client|service
```

Comment:
AFAIR, we also support additional flags to set dependent proto paths. @MadhukaHarith92 can you point the fix

Reply:
https://github.com/ballerina-platform/ballerina-standard-library/issues/1612

Reply:
Added in here https://github.com/ballerina-platform/module-ballerina-grpc/pull/583/commits/3a14d279470bf270274bfbe4cffa1b18de3407cb